app/views/projects/show.html.erb

```diff
@@ -52,6 +53,24 @@
     $("form").on('ajax:error', function(event, data, status, xhr) {
       $(this).addClass('error-saving');
     });
+
+    var contentItemTitlesToFormElements = {}
+    $(".edit_project_content_item").each(function(index, formElement) {
+      var $formElement = $(formElement);
+      var titleAnchor = $formElement.find("a")[0];
+      contentItemTitlesToFormElements[titleAnchor.innerHTML.toLowerCase()] = $formElement;
+    });
+
+    $("#project-tagging-content-item-search").keyup(function(event) {
```

Comment:
It just filters the data already on the page, hiding any elements that don't match the search, and showing the rest.